src/jwe/encrypt.rs

```diff
@@ -0,0 +1,149 @@
+use askar_crypto::{
+    buffer::SecretBytes,
+    encrypt::{KeyAeadInPlace, KeyAeadMeta},
+    jwk::ToJwk,
+    kdf::{FromKeyDerivation, KeyExchange},
+    repr::{KeyGen, ToSecretBytes},
+};
+use serde_json::Value;
+use sha2::{Digest, Sha256};
+
+use crate::{
+    error::{ErrorKind, Result, ResultExt},
+    jwe::envelope::{Algorithm, EncAlgorithm, PerRecipientHeader, ProtectedHeader, Recepient, JWE},
+    utils::crypto::{JoseKDF, KeyWrap},
+};
+
+pub(crate) fn encrypt<CE, KDF, KE, KW>(
+    plaintext: &[u8],
+    alg: Algorithm,
```

Comment:
Do we check that only A256CBC is supported for ECDH-1PU?

Reply:
It will be forced on message::pack / message::unpack level. These function isn't aware about any concrete crypto.